docker/pulsar/Dockerfile

```diff
@@ -76,8 +86,8 @@ RUN pip3 install pyyaml==5.4.1
 # 4. /pulsar - hadoop writes to this directory
 RUN mkdir /pulsar && chmod g+w /pulsar
 
-ENV JAVA_HOME /usr/lib/jvm/java-17-openjdk-amd64
-RUN echo networkaddress.cache.ttl=1 >> /usr/lib/jvm/java-17-openjdk-amd64/conf/security/java.security
+ENV JAVA_HOME /usr/lib/jvm/temurin-17-jdk-amd64
+RUN echo networkaddress.cache.ttl=1 >> /usr/lib/jvm/temurin-17-jdk-amd64/conf/security/java.security
```

Comment:
Before this change, when I build on M1 with `Docker version 20.10.17, build 100c701`, it failed with:

```
[INFO] DOCKER> Step 14/25 : ENV JAVA_HOME /usr/lib/jvm/java-17-openjdk-amd64
[INFO] DOCKER> 
[INFO] DOCKER> ---> Running in a47a209de148
[INFO] DOCKER> Removing intermediate container a47a209de148
[INFO] DOCKER> ---> 6a9d5c8543b8
[INFO] DOCKER> Step 15/25 : RUN echo networkaddress.cache.ttl=1 >> /usr/lib/jvm/java-17-openjdk-amd64/conf/security/java.security
[INFO] DOCKER> 
[INFO] DOCKER> ---> Running in d55c15e90ed1
[INFO] DOCKER> [91m/bin/sh: 1: cannot create /usr/lib/jvm/java-17-openjdk-amd64/conf/security/java.security: Directory nonexistent

[INFO] DOCKER> Removing intermediate container d55c15e90ed1
[ERROR] DOCKER> Unable to build image [apachepulsar/java-test-image] : "The command '/bin/sh -c echo networkaddress.cache.ttl=1 >> /usr/lib/jvm/java-17-openjdk-amd64/conf/security/java.security' returned a non-zero code: 2"  ["The command '/bin/sh -c echo networkaddress.cache.ttl=1 >> /usr/lib/jvm/java-17-openjdk-amd64/conf/security/java.security' returned a non-zero code: 2" ]
```

Perhaps it's unrelated to this patch. Just throw here by any chances we may `mkdir -p` before `echo`

Reply:
After add:

```
RUN mkdir -p /usr/lib/jvm/java-17-openjdk-amd64/conf/security
```

this issue vanished.

I may submit a patch to add it after you migrate openjdk to temurin. `mkdir -p` should not do harm.

Reply:
Could you try using the Temurin JDK on the M1?

Reply:
To build docker image on M1, I need a preceding patch (https://github.com/apache/pulsar/pull/17148). You can help with reviewing that patch and I'd like to verify this one integrated with it.

Reply:
It seems the first try gives:

```
[INFO] DOCKER> deb [signed-by=/etc/apt/keyrings/adoptium.asc] https://packages.adoptium.net/artifactory/deb focal main
[INFO] DOCKER> Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
[INFO] DOCKER> Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
[INFO] DOCKER> Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
[INFO] DOCKER> Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
[INFO] DOCKER> Ign:5 https://packages.adoptium.net/artifactory/deb focal InRelease
[INFO] DOCKER> Err:6 https://packages.adoptium.net/artifactory/deb focal Release
  Could not wait for server fd - select (11: Resource temporarily unavailable) [IP: 198.19.20.71 443]
[INFO] DOCKER> Reading package lists...
[INFO] DOCKER> 
[INFO] DOCKER> [91mE: The repository 'https://packages.adoptium.net/artifactory/deb focal Release' does not have a Release file.

[INFO] DOCKER> Removing intermediate container 322d69692fb0
[ERROR] DOCKER> Unable to build image [apachepulsar/java-test-image] : "The command '/bin/sh -c mkdir -p /etc/apt/keyrings      && wget -O - https://packages.adoptium.net/artifactory/api/gpg/key/public | tee /etc/apt/keyrings/adoptium.asc      && echo \"deb [signed-by=/etc/apt/keyrings/adoptium.asc] https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main\" | tee /etc/apt/sources.list.d/adoptium.list      && apt-get update      && apt-get -y dist-upgrade      && apt-get -y install temurin-17-jdk' returned a non-zero code: 100"  ["The command '/bin/sh -c mkdir -p /etc/apt/keyrings      && wget -O - https://packages.adoptium.net/artifactory/api/gpg/key/public | tee /etc/apt/keyrings/adoptium.asc      && echo \"deb [signed-by=/etc/apt/keyrings/adoptium.asc] https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main\" | tee /etc/apt/sources.list.d/adoptium.list      && apt-get update      && apt-get -y dist-upgrade      && apt-get -y install temurin-17-jdk' returned a non-zero code: 100" ]
```